src/components/Table/_pagination.scss

```diff
@@ -23,3 +23,9 @@
     margin-left: 0rem;
   }
 }
+
+// This is a workaround to hide the select controls since the
+// page size select can't be disabled.
+.#{$iot-prefix}--pagination--hide-select .bx--select {
+  display: none;
```

Comment:
Page size select can be disabled via [the `pageInputDisabled` prop on Pagination](https://github.com/carbon-design-system/carbon/blob/468d6ac9a524b031f920bcb8048b8c4e3e865823/packages/react/src/components/Pagination/Pagination.js#L126-L129) from Carbon

Reply:
Unless this has been fixed in carbon during the last days, the problem is that the `pageInputDisabled` prop on Pagination only disables the input for selecting a predefined page, it does not disable the input for defining a new page size. If the use selects a new page size new data might be asynchronously loaded, which should be prevented here. I consider the fact that the pagination can't be completely disabled a bug and I've filed an [issue](https://github.com/carbon-design-system/carbon/issues/6055) for that.